src/biogeophys/SnowCoverFractionMod.F90

```diff
@@ -0,0 +1,785 @@
+       urbpoi, h2osno_total, snowmelt, int_snow, newsnow, bifall, &
+       snow_depth, frac_sno)
+    !
+    ! !DESCRIPTION:
+    ! Update snow depth and snow fraction using the NY07 parameterization
+    !
+    ! !ARGUMENTS:
+    class(snow_cover_fraction_ny07_type), intent(in) :: this
+    type(bounds_type), intent(in) :: bounds
+    integer, intent(in) :: num_c       ! number of columns in filter_c
+    integer, intent(in) :: filter_c(:) ! column filter to operate over
+
+    logical  , intent(in)    :: urbpoi( bounds%begc: )       ! true if the given column is urban
+    real(r8) , intent(in)    :: h2osno_total( bounds%begc: ) ! total snow water (mm H2O)
+    real(r8) , intent(in)    :: snowmelt( bounds%begc: )     ! total snow melt in the time step (mm H2O)
+    real(r8) , intent(in)    :: int_snow( bounds%begc: )     ! integrated snowfall (mm H2O)
+    real(r8) , intent(in)    :: newsnow( bounds%begc: )      ! total new snow in the time step (mm H2O)
+    real(r8) , intent(in)    :: bifall( bounds%begc: )       ! bulk density of newly fallen dry snow (kg/m3)
+
+    real(r8) , intent(inout) :: snow_depth( bounds%begc: )   ! snow height (m)
+    real(r8) , intent(inout) :: frac_sno( bounds%begc: )     ! fraction of ground covered by snow (0 to 1)
+    !
+    ! !LOCAL VARIABLES:
+    integer :: fc, c
+
+    character(len=*), parameter :: subname = 'UpdateSnowDepthAndFracNY07'
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_FL((ubound(urbpoi, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(h2osno_total, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(snowmelt, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(int_snow, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(newsnow, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(bifall, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(snow_depth, 1) == bounds%endc), sourcefile, __LINE__)
+    SHR_ASSERT_FL((ubound(frac_sno, 1) == bounds%endc), sourcefile, __LINE__)
+
+    do fc = 1, num_c
+       c = filter_c(fc)
+
+       if (h2osno_total(c) > 0.0) then
+          snow_depth(c) = snow_depth(c) + newsnow(c) / bifall(c)
+          if(snow_depth(c) > 0.0_r8)  then
+             frac_sno(c) = tanh(snow_depth(c) / (2.5_r8 * this%zlnd * &
+                  (min(800._r8,(h2osno_total(c)+ newsnow(c))/snow_depth(c))/100._r8)**1._r8) )
+          end if
+          if(h2osno_total(c) < 1.0_r8)  then
+             frac_sno(c)=min(frac_sno(c),h2osno_total(c))
+          end if
+
+       else !h2osno_total == 0
+          ! initialize frac_sno and snow_depth when no snow present initially
+          if (newsnow(c) > 0._r8) then
+             snow_depth(c) = newsnow(c) / bifall(c)
+             if(snow_depth(c) > 0.0_r8)  then
+                frac_sno(c) = tanh(snow_depth(c) / (2.5_r8 * this%zlnd * &
```

Comment:
There are a couple magic numbers that could be made parameters here.

Reply:
I'd rely on scientists (@swensosc 	or others) to provide meaningful names for the parameters.